### PR TITLE
Update save/load print statements for FSDP benchmark

### PR DESCRIPTION
**Pull request by MattIrv**
Now it prints each save/load time individually and also all at the end. This makes it easier to understand how the data differs from sample to sample.

### CODE REVIEW

**Review by abhibyreddi — APPROVED** [review]

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py [hunk 1]
```diff
@@ -186,12 +192,17 @@ def main(ckpt_dir_path: str, ckpt_restore_path: str = ""):
...
+            print(
+                f"Loaded checkpoint from {new_ckpt_dir_path} in {end - start} seconds"
+            )
         load_checkpoint_times.append(end - start)
 
     if torch.distributed.get_rank() == 0:
         avg_load_time = statistics.mean(load_checkpoint_times)
+        avg_save_time = statistics.mean(save_checkpoint_times)
         print_times(args, avg_save_time, avg_load_time)
+        print(f"All save times: {save_checkpoint_times}")
```
Comment (jdnurme) [inline dataflux_pytorch/benchmark/checkpointing/multinode/train.py:204]: nit: we could just put `if args.save_only:` around the print for loads to make output cleaner in that case.

**Review by jdnurme — APPROVED** [review]

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py [hunk 2]
```diff
@@ -142,14 +143,19 @@ def main(ckpt_dir_path: str, ckpt_restore_path: str = ""):
     )
     trainer.fit(model, dataloader)
     print(f"Saving checkpoint to {ckpt_dir_path} {num_save_calls} times.")
-    start = time.time()
+    save_checkpoint_times = []
```
Comment (awonak) [inline dataflux_pytorch/benchmark/checkpointing/multinode/train.py:146]: Also consider using [torch.cuda.Event()](https://pytorch.org/docs/stable/generated/torch.cuda.Event.html) to `record()` and measure `elapsed_time()`. 

See an example usage in section 2.4 of https://pytorch.org/tutorials/intermediate/FSDP_tutorial.html#how-to-use-fsdp

Whatever you decide here, I'll update my PR #168 to match.

**Review by awonak — APPROVED** [review]

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py (hunk 2 above)
Comment (MattIrv) [reply dataflux_pytorch/benchmark/checkpointing/multinode/train.py:146]: Our calls to save/load checkpoints here already enforce synchronization at the end in the underlying pytorch distributed logic, so I don't think I need to do anything special. Your case might be different if the nodes aren't already synchronized at that point.

File: dataflux_pytorch/benchmark/checkpointing/multinode/train.py (hunk 1 above)
Comment (MattIrv) [reply dataflux_pytorch/benchmark/checkpointing/multinode/train.py]: Done

**Pull request merged**